pyogrio/_ogr.pyx

```diff
@@ -122,6 +123,30 @@ def ogr_list_drivers():
     return drivers
 
 
+def buffer_to_virtual_file(bytesbuf, ext=''):
+    """Maps a bytes buffer to a virtual file.
+    `ext` is empty or begins with a period and contains at most one period.
+
+    This (and remove_virtual_file) is originally copied from the Fiona project
+    (https://github.com/Toblerity/Fiona/blob/c388e9adcf9d33e3bb04bf92b2ff210bbce452d9/fiona/ogrext.pyx#L1863-L1879)
+    """
+
+    vsi_filename = '/vsimem/{}'.format(uuid4().hex + ext)
+
+    vsi_handle = VSIFileFromMemBuffer(vsi_filename.encode("utf8"), <unsigned char *>bytesbuf, len(bytesbuf), 0)
```

Comment:
Elsewhere we usually handle Python => C strings in multiple steps, I thought in part because not doing so triggers a [compilation error](https://cython.readthedocs.io/en/stable/src/tutorial/strings.html#encoding-text-to-bytes).  And we've standarded on using `"UTF-8"` to refer to unicode in Cython.

So this would be 
```
char *filename_c = NULL
filename_b = vsi_filename.encode("UTF-8")
filename_c = filename_b
vsi_handle = VSIFileFromMemBuffer(filename_c, <unsigned char *>bytesbuf, len(bytesbuf), 0)
```

Though I'm not sure that is strictly necessary.  (same for `remove_virtual_filename` too)

Reply:
It doesn't seem to be necessary, since it is compiling here?

(but already changed utf8 to UTF-8)